packages/shared/hooks/useAsync.ts

```diff
@@ -107,6 +124,28 @@ export function useAsync<R, T extends Function>(cb?: AsyncCb<R, T>) {
   return [state, run, setAttempt] as const;
 }
 
+function useIsMounted() {
+  const isMounted = useRef(false);
+
+  useEffect(() => {
+    isMounted.current = true;
+
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+
+  return useCallback(() => isMounted.current, []);
+}
+
+// TODO: Rename the error.
+export class CanceledError extends Error {
```

Comment:
A bit of a nit but is this really an error? As this can happen via the normal successful operation of the code it feels more like a status...

Reply:
Hmmm... I see your point, but I think it was named this way to think of this is "did the `run` function complete". If we abort early, it means that "no the `run` function DIDNT complete... because we aborted it", rather than asking if the request (or whatever cb) completed. A case could be made either way but, I'm for keeping it as is to have a 'consistent' api. Thoughts?